toxcore/group_chats.c

```diff
@@ -7034,27 +7035,24 @@ static void do_gc_tcp(const GC_Session *c, GC_Chat *chat, void *userdata)
  * Updates our TCP and UDP connection status and flags a new announcement if our connection has
  * changed and we have either a UDP or TCP connection.
  */
-#define GC_SELF_CONNECTION_CHECK_INTERVAL 2
+#define GC_SELF_CONNECTION_CHECK_INTERVAL 5
+#define GC_SELF_REFRESH_ANNOUNCE_INTERVAL (60 * 20)
 non_null()
 static void do_self_connection(const GC_Session *c, GC_Chat *chat)
 {
     if (!mono_time_is_timeout(chat->mono_time, chat->last_self_announce_check, GC_SELF_CONNECTION_CHECK_INTERVAL)) {
         return;
     }
 
-    const uint32_t tcp_connections = tcp_connected_relays_count(chat->tcp_conn);
-
-    const Messenger *m = c->messenger;
-    const unsigned int self_udp_status = ipport_self_copy(m->dht, &chat->self_ip_port);
-
+    const unsigned int self_udp_status = ipport_self_copy(c->messenger->dht, &chat->self_ip_port);
     const bool udp_change = (chat->self_udp_status != self_udp_status) && (self_udp_status != SELF_UDP_STATUS_NONE);
-    const bool tcp_change = tcp_connections != chat->tcp_connections;
 
-    if (is_public_chat(chat) && (udp_change || tcp_change)) {
+    if (is_public_chat(chat) &&
```

Comment:
This condition is quite hard to parse, can you comment/simplify it?

Reply:
Done.